kubeadm/v1.15.0/KubeCluster.ps1

```diff
@@ -0,0 +1,321 @@
+$hnsPath = "https://raw.githubusercontent.com/Microsoft/SDN/master/Kubernetes/windows/hns.psm1"
+$hnsDestination = "$PSScriptRoot\hns.psm1" 
+DownloadFile -Url $hnsPath -Destination $hnsDestination
+Import-Module $hnsDestination
+
+
+ReadKubeclusterConfig -ConfigFile $ConfigFile
+InitHelper
+PrintConfig
+WriteKubeClusterConfig
+
+
+# Initialize internal network modes of windows corresponding to 
+# the plugin used in the cluster
+$Global:NetworkName = "cbr0"
+$Global:NetworkMode = "l2bridge"
+if ($Global:NetworkPlugin -eq "vxlan")
+{
+    $Global:NetworkMode = "overlay"
+    $Global:NetworkName = "vxlan0"
+}
+######################################################################################################################
+
+# Handle --install
+function
+Restart-And-Run()
+{
+    Write-Output "Restart is required; restarting now..."
+
+    $argList = $script:MyInvocation.Line.replace($script:MyInvocation.InvocationName, "")
+
+    #
+    # Update .\ to the invocation directory for the bootstrap
+    #
+    $scriptPath = $script:MyInvocation.MyCommand.Path
+
+    $argList = $argList -replace "\.\\", "$pwd\"
+
+    if ((Split-Path -Parent -Path $scriptPath) -ne $pwd)
+    {
+        $sourceScriptPath = $scriptPath
+        $scriptPath = "$pwd\$($script:MyInvocation.MyCommand.Name)"
+
+        Copy-Item $sourceScriptPath $scriptPath
+    }
+
+    Write-Output "Creating scheduled task action ($scriptPath $argList)..."
+    $action = New-ScheduledTaskAction -Execute "powershell.exe" -Argument "-NoExit $scriptPath $argList"
+
+    Write-Output "Creating scheduled task trigger..."
+    $trigger = New-ScheduledTaskTrigger -AtLogOn
+
+    Write-Output "Registering script to re-run at next user logon..."
+    Register-ScheduledTask -TaskName $global:BootstrapTask -Action $action -Trigger $trigger -RunLevel Highest | Out-Null
+
+    try
+    {
+        if ($Force)
+        {
+            Restart-Computer -Force
+        }
+        else
+        {
+            Restart-Computer
+        }
+    }
+    catch
+    {
+        Write-Error $_
+
+        Write-Output "Please restart your computer manually to continue script execution."
+    }
+
+    exit
+}
+
+if ($install.IsPresent)
+{
+    if (InstallContainersRole)
+    {
+        $res = Read-Host "Continue to Reboot the host [Y/n] - Default [Y]"
+        if ($res -eq '' -or $res -eq 'Y'  -or $res -eq 'y')
+        {
+            Restart-And-Run
+        }
+    }
+
+    if (!(Test-Path $env:HOMEDRIVE/$env:HOMEPATH/.ssh/id_rsa.pub))
+    {
+        $res = Read-Host "Do you wish to generate a SSH Key & Add it to the Linux control-plane node [Y/n] - Default [Y]"
+        if ($res -eq '' -or $res -eq 'Y'  -or $res -eq 'y')
+        {
+            ssh-keygen.exe
+        }
+    }
+
+    $pubKey = Get-Content $env:HOMEDRIVE/$env:HOMEPATH/.ssh/id_rsa.pub
+    Write-Host "Execute the below commands on the Linux control-plane node ($Global:MasterIp) to add this Windows node's public key to its authorized keys"
```

Comment:
Can we move the ssh stuff to line 216 at the end of the install procedure? The output from `InstallCRI` overwrites this txt quickly and there are no pauses so users may overlook that this is an action they must take :)